src/lib/Libifl/pbsD_connect.c

```diff
@@ -1200,11 +1199,94 @@ get_svr_inst_fd(int vfd, char *svr_inst_id)
 
 	if (svr_inst_name == NULL)
 		return -1;
-		
+
 	for (i = 0; svr_conns[i]; i++) {
 		if (is_same_host(svr_conns[i]->name, svr_inst_name) && (svr_inst_port == svr_conns[i]->port))
 			return svr_conns[i]->sd;
 	}
 
 	return -1;
 }
+
+/**
+ * @brief	Helper function to convert a server name to corresponding fd
+ *
+ * @param[in]	c - Cluster/virtual connection descriptor
+ * @param[in]	svrname - name of the server instance to get fd for
+ *
+ * @return	int
+ * @retval	fd of server
+ * @retval	-1 if not found
+ */
+static int
+server_name_to_fd(int c, char *svrname)
+{
+	svr_conn_t **conns = get_conn_svr_instances(c);
+	int i;
+
+	if (conns == NULL) {
+		pbs_errno = PBSE_NOCONNECTS;
+		return -1;
+	}
+	for (i = 0; i < get_num_servers(); i++) {
+		if (strcmp(conns[i]->name, svrname) == 0)
+			return conns[i]->sd;
+	}
+
+	return -1;
+}
+
+/**
+ * @brief	Get the server fd associated with a jobid
+ * @param[in]	c - virtual fd of the cluster
+ * @param[in]	jobid - id of the job
+ *
+ * @return	int
+ * @retval	fd of the server instance
+ * @retval	-1 for error
+ */
+int
+get_server_fd_from_jid(int c, char *jobid)
+{
+	char server_out[PBS_MAXSERVERNAME];
+	char job_id_out[PBS_MAXCLTJOBID];
+	int fd = -1;
+	static char *dflt_server = NULL;
+
+	server_out[0] = '\0';
+
+	if (dflt_server == NULL)
+		dflt_server = pbs_default();
+
+	/* Until we resolve jobid namespace for multi-server, each multi-server
+	 * will have a unique PBS_SERVER value, so server id can be derived from the jobid itself
+	 */
+	get_server(jobid, job_id_out, server_out);
+	if (server_out[0] == '\0' && dflt_server != NULL)
+		pbs_strncpy(server_out, dflt_server, sizeof(server_out));
+
+	if (server_out[0] != '\0')
+		fd = server_name_to_fd(c, server_out);
+
+	return fd;
+}
+
+/**
+ * @brief	To be used by IFL to check whether it should talk to multiple servers
+ *
+ * @param[in]	fd - the connection descriptor to check
+ *
+ * @return int
+ * @retval 1 for yes, 0 for no
+ */
+int
+multi_svr_op(int fd)
+{
+	svr_conn_t **conns = get_conn_svr_instances(fd);
+	int num_svrs = get_num_servers();
```

Comment:
Check the length of the array instead. If you want to check its single instance cluster, you can do `fd == conns[0]->fd`